### PR TITLE
examples/ct: sort leaves by LeafIndex

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

File: examples/ct/handlers_test.go [hunk 1]
```diff
@@ -864,43 +846,89 @@ func TestGetEntriesRejectsMissingParams(t *testing.T) {
...
 			t.Fatal(err)
 		}
-
 		w := httptest.NewRecorder()
 		handler.ServeHTTP(w, req)
 
-		if expected, got := testCase.expectedStatus, w.Code; expected != got {
-			t.Fatalf("expected status %d, got %d for test case %s", expected, got, testCase.explanation)
+		if got := w.Code; got != test.want {
+			t.Errorf("GetEntries(%d, %d)=%d; want %d for test  %s", test.start, test.end, got, test.want, test.desc)
```
Comment (AlCutter) [inline examples/ct/handlers_test.go]: spurious space before final %s

File: examples/ct/handlers_test.go [hunk 2]
```diff
@@ -864,43 +846,89 @@ func TestGetEntriesRejectsMissingParams(t *testing.T) {
...
+		} else if test.rpc && !strings.Contains(w.Body.String(), "RPCMADE") {
```
Comment (AlCutter) [inline examples/ct/handlers_test.go]: You could separate these if tests and potentially pop a `continue` in the one above if you don't want to, er, continue when the RPC wasn't made?

File: examples/ct/handlers_test.go [hunk 3]
```diff
@@ -864,43 +846,89 @@ func TestGetEntriesRejectsMissingParams(t *testing.T) {
...
-		// chance
-		if testCase.expectedStatus == http.StatusInternalServerError {
-			if !strings.Contains(w.Body.String(), "RPCMADE") {
-				t.Fatalf("Did not get expected backend error: %s\n%s", testCase.explanation, w.Body)
+func TestSortLeafRange(t *testing.T) {
+	var tests = []struct {
+		start   int64
+		end     int64
+		entries []int
+		errstr  string
```
Comment (AlCutter) [inline examples/ct/handlers_test.go]: errStr maybe?

File: examples/ct/handlers_test.go [hunk 4]
```diff
@@ -864,43 +846,89 @@ func TestGetEntriesRejectsMissingParams(t *testing.T) {
...
+		rsp := trillian.GetLeavesByIndexResponse{}
+		for _, idx := range test.entries {
+			rsp.Leaves = append(rsp.Leaves, &trillian.LogLeaf{LeafIndex: int64(idx)})
+		}
+		err := sortLeafRange(&rsp, test.start, test.end)
+		if test.errstr != "" {
+			if err == nil {
+				t.Errorf("sortLeafRange(%v, %d, %d)=nil; want %q", test.entries, test.start, test.end, test.errstr)
+			} else if !strings.Contains(err.Error(), test.errstr) {
+				t.Errorf("sortLeafRange(%v, %d, %d)=%v; want %q", test.entries, test.start, test.end, err, test.errstr)
```
Comment (AlCutter) [inline examples/ct/handlers_test.go]: Maybe say "want substring %d" (or similar)?

File: examples/ct/handlers_test.go [hunk 5]
```diff
@@ -985,7 +1013,7 @@ func TestGetEntriesBackendReturnedNonContiguousRange(t *testing.T) {
 	if got, want := w.Code, http.StatusInternalServerError; got != want {
 		t.Fatalf("expected %v for backend too many leaves, got %v. Body: %v", want, got, w.Body)
 	}
-	if in, want := w.Body.String(), "non contiguous"; !strings.Contains(in, want) {
+	if in, want := w.Body.String(), "unexpected leaf index"; !strings.Contains(in, want) {
```
Comment (AlCutter) [inline examples/ct/handlers_test.go]: same here

File: examples/ct/handlers_test.go (hunk 1 above)
Comment (daviddrysdale) [inline examples/ct/handlers_test.go]: Done.

File: examples/ct/handlers_test.go (hunk 2 above)
Comment (daviddrysdale) [inline examples/ct/handlers_test.go]: I've split the tests but not done a `continue` (which would jump over the `mockCtrl.Finish()`.

File: examples/ct/handlers_test.go (hunk 3 above)
Comment (daviddrysdale) [inline examples/ct/handlers_test.go]: Done.

File: examples/ct/handlers_test.go (hunk 4 above)
Comment (daviddrysdale) [inline examples/ct/handlers_test.go]: Done.

**Pull request merged**